packages/hadron-app-registry/src/register-plugin.tsx

```diff
@@ -144,11 +144,90 @@ const useMockOption = <T extends keyof MockOptions>(key: T): MockOptions[T] => {
   return useContext(MockOptionsContext)[key];
 };
 
+function useHadronPluginActivate<T, S extends Record<string, () => unknown>>(
```

Comment:
The logic here is exactly the same as before, just moved to a hook outside the component so that it can be re-exported through the plugin interface